homeassistant/helpers/entity.py

```diff
@@ -24,13 +24,20 @@ def generate_entity_id(entity_id_format: str, name: Optional[str],
                        current_ids: Optional[List[str]]=None,
                        hass: Optional[HomeAssistant]=None) -> str:
     """Generate a unique entity ID based on given entity IDs or used IDs."""
-    name = (name or DEVICE_DEFAULT_NAME).lower()
     if current_ids is None:
         if hass is None:
             raise ValueError("Missing required parameter currentids or hass")
 
         current_ids = hass.states.entity_ids()
 
+    return async_generate_entity_id(entity_id_format, name, current_ids)
+
+
+def async_generate_entity_id(entity_id_format: str, name: Optional[str],
```

Comment:
Why we need this function as async? The normal `generate_entity_id` call this without thread_safe and I see no other things they will that calling.